project/WelshTranslation.scala

```diff
@@ -0,0 +1,194 @@
+            }
+          case None =>
+            line
+        }
+      }
+
+    val newWelshLanguageFile = newWelshLines.mkString(System.lineSeparator()).getBytes(StandardCharsets.UTF_8)
+
+    val changedTranslations = gitDiffLanguageFileChanges()
+      .map { change => s"${change.translation.key},${change.translation.text}"}
+
+    val translationTeamFile = (newlinesToTranslate.toList ::: changedTranslations).mkString(System.lineSeparator()).getBytes(StandardCharsets.UTF_8)
+
+    Files.write(Paths.get("conf/NEW_WELSH_messages.cy"), newWelshLanguageFile)
+    Files.write(Paths.get("conf/CdsReimbursementNewOrChanged.csv"), translationTeamFile)
+
+    ()
+  }
+
+  def gitDiffLanguageFileChanges(): List[LineChange] = {
+    val gitLog = "git --no-pager log -30".!!
+    val commits = """commit (\w+)""".r //Commit log lines: commit 043c494a3058d69e055be485bd30e485eb04cc9d
+    val comitIds = commits.findAllMatchIn(gitLog).map(_.group(1)).toList
+
+    val changedLineRegex = "^[-+](?![-+])(.+)".r //exclude lines like these: --- a/conf/messages OR +++ b/conf/messages
+
+    comitIds
+      .zipWithIndex
+      .zip(comitIds.tail)
+      .map { case ((newCommit, date), oldCommit) =>
+        val gitDiff = s"git diff $oldCommit $newCommit -U0 conf/messages"
+        println(gitDiff)
+        gitDiff
+          .!!
+          .split(System.lineSeparator())
+          .toList
+          .map(changedLineRegex.findFirstIn(_)) //Keep only git changed lines
+          .flatten(Option.option2Iterable)
+          .map(parseGitMessagesFileLine(_, date))
+          .flatten(Option.option2Iterable)
+          .filter(_.operation == Addition)
+      }
+      .flatten
+      .groupBy(_.translation.key)
+      .toList
+      .map(_._2.sortWith((a, b) => a.date < b.date).head)
+  }
+
+  def parseGitMessagesFileLine(line: String, date: Int): Option[LineChange] = {
+    parseMessagesFileLine(line.tail) match {
+      case Some(translation: Translation) =>
+        (line(0) == '+') match {
+          case true =>
+            Some(LineChange(date, Addition, translation))
+          case false =>
+            Some(LineChange(date, Removal, translation))
+        }
+      case None =>
+        None
+    }
+  }
+
+  def parseMessagesFileLine(line: String): Option[Translation] = {
+    val separator = line.indexOf('=')
+    val sectionSeparator = line.indexOf("#==")
+    (separator >= 0 && sectionSeparator < 0) match {
+      case true =>
+        val languageKey = line.substring(0, separator)
+        val text = line.substring(separator + 1)
+        Some(Translation(languageKey, text))
+      case false =>
+        None
+    }
+  }
+
+  def linesToMap(lines: List[String]): Map[String, String] = {
+    val kvs = lines
+      .map { line =>
+        val separator = line.indexOf('=')
+        if (separator >= 0 && (line.length > separator + 1)) {
+          Some(line.substring(0, separator) -> line.substring(separator + 1))
+        } else
+          None
```

Comment:
Missing Scala 2.13 - You can do Option.when(predicate)(whenTrue: => ???) which is handy